Automattic/jetpack
# Refactor full sync chunking to not repeat unnecessary DB queries and processing

This PR aims to simplify the while loop [here](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/sync/src/modules/class-module.php#L342)
so that we don't need to make unnecessary calls to the DB or any processing that might happen in `get_next_chunk`. The simple scenario would be when the limit of max chunks has been reached.

## Proposed changes:

The simple approach of checking for the condition at the end of the loop is not valid since setting the status to finished is part of the logic and needs to be taken into consideration. 
A refactor of the loop that improves the efficiency while making the code more readable is suggested. 

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Let's start with a self-hosted site with this branch applied (be it your local or a JN)
* Create a bunch of posts (more than 100). I suggest using some plugin to easily duplicate.
* Create also some comments.
* Go to Jetpack Debugger and Schedule a Full Sync only.
* Verify the content was synced by checking the corresponding sync-audit logs (link provided in the Jetpack Debugger but change the queue to immediate-send)
